include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -281,11 +281,126 @@ namespace RINGMesh
          * @return the attribute manager containing the grid point attributes
          * values.
          */
-        GEO::AttributesManager& attributes_manager() const
+        GEO::AttributesManager& attributes_manager()
         {
             return attributes_manager_;
         }
 
+        /*!
+         * @return the attribute manager containing the grid point attributes
+         * values.
+         */
+        const GEO::AttributesManager& attributes_manager() const
+        {
+            return attributes_manager_;
+        }
+
+        /*!
+         * Allows us to avoid code duplication on the attribute creation
```

Comment:
I suggest to use a map instead of "if... else if" blocks. You can also update the documentation saying what the map does (instead of explaining why you implement it)

Reply:
We can't do a map between a type and a string, the map key is of a definite type.